File: 2 semestre/Técnica de Desenvolvimento de Algoritmos/Listas de Exercicios/Lista 4/ex15.c
#include <stdio.h>

main()
{
    printf("Lendo os indices de um vetor\n\n");

    //vars
    int v[9], i=0, cont=0;

    //Entrada
    while( cont < 9)
    {
        printf("Digite v[%d] ( < 9 ): ", i);
        scanf("%d", &v[i]);

        //Processamento
        i = v[i];
        cont++;
    }

    //Saída
    printf("\nVetor Resultante: \n0  ");
    for( i=0; i<9; i++)
    {
        if( v[i] != 9)
            printf("%d  ", v[i]);
    }

}

File: 2 semestre/Laboratório de Desenvolvimento de Algoritmos/Lista de Exercícios 4/src/Ex15.java
//Importando Scanner
import java.util.Scanner;

public class Ex15 {

	public static void main(String[] args){
		
		//Instanciando Scanner
		Scanner ent = new Scanner(System.in);
		
		System.out.println("Valor do próximo elemento a ser lido\n\n");
		
		//vars
		int v[] = new int[9];
		int cont=0, i=0;
		
		
		while( cont<9)
		{
			//Entrada
			System.out.print("Digite v[" + i + "] (<9): ");
			v[i] = ent.nextInt();
			
			//Processamento
			i = v[i];
			cont++;
		}
		
		//Saída
		System.out.println("Vetor Resultante: ");
		for( i=0; i<9; i++)
		{
			System.out.printf("%d  ", v[i]);
		}
		
		
	}
}
